### ISSUE TITLE
1.16.5 Rendering issue with HD_U_G8_pre12 + shaders + Refined Storage

### ISSUE THREAD

**Issue by bmwalter68** [sp614x/optifine#5606]
## Description of Issue
The issue arose between HD_U_G8_pre2 and HD_U_G8_pre3.  Symptoms include diagonal colored lines across the faces of Refined Storage blocks using cutout models.  Interestingly, the issue was present on *ALL* blocks in pre3 but was limited to just Refined Storage blocks by pre5 (I don't have pre4 to test on)  The timeline is:

pre1: No rendering issue
pre2: No rendering issue but extremely long load time
pre3: Rendering issue on all blocks
pre5-pre12: Rendering issue limited to cutout block models from Refined Storage

## Steps to Reproduce
1. Create an MC 1.16.5 + Forge 36.1.13 instance in CurseForge
2. Install OptiFine + OptiForge + Refined Storage + shaders (have tested with SEUS Renewed, Slidurs, Complimentary and Sushi)
3. Rendering issues present for Refined Storage cutout model blocks

## OptiFine Version
1.16.5_HD_U_G8_pre3 through pre12

## Installation Method
Created instance with CurseForge and dropped preview mod jar files into mods directory.

## Fabric/Forge Version
Forge 36.1.13

## OptiForge Version
pre1-pre8: OptiForge-MC1.16.5-0.5.4
pre9-pre12: OptiForge-MC1.16.5-0.5.5

## Other Installed Mods
jei-1.16.5-7.6.4.87
journeymap-1.16.5-5.7.1
refinedstorage-1.9.12
refinedstorageaddons-0.7.2
(tested positive both with and without journeymap)

## Log Files/Crash Reports
No crash, just rendering issues

## F3 Debug Screenshot
![2021-04-29_11 20 48](https://user-images.githubusercontent.com/4651048/116575886-11837a00-a8dd-11eb-922e-78bf94a154bd.png)

## Prior Testing
I can't be sure if other mods suffer from this issue, but outside of pre3 where the rendering issue affected all blocks I've only seen it affect Refined Storage cutout models.

## Additional Information
Screenshots of the issue:

pre1: no rendering issues
![pre1_2021-04-29_09 23 57](https://user-images.githubusercontent.com/4651048/116576010-2c55ee80-a8dd-11eb-95e3-4030d7870c27.png)

pre3: all blocks
![pre3_2021-04-29_09 21 59](https://user-images.githubusercontent.com/4651048/116576081-3d066480-a8dd-11eb-96fe-2362a3afb14b.png)

pre5-pre12: refined storage cutout model blocks
![pre12_2021-04-29_09 26 27](https://user-images.githubusercontent.com/4651048/116576153-4d1e4400-a8dd-11eb-9453-209c9b1c6432.png)

pre12: no texture pack
![pre12_no_textures_2021-04-29_09 27 03](https://user-images.githubusercontent.com/4651048/116576351-7b9c1f00-a8dd-11eb-99b2-0cd7b31d27be.png)

**Comment by KaiAF** [comment]
Have you tried not using optiforge? You can put optifine into the mods folder. I don't think there is an actual need to use optiforge to be honest.

**Comment by bmwalter68** [comment]
My understanding was that OptiForge was required to properly disable the Forge lighting pipeline for shaders.  Is this not the case?

**Comment by KaiAF** [comment]
I honestly have no clue what optiforge is for

**Comment by HiItsDevin** [comment]
If I recall correctly, OptiForge was used during times where OptiFine itself can't use a certain Forge version, so OptiForge would be able to use OptiFine with more Forge versions?

tl;dr compatibility layer when OptiFine's forge layer didn't work

**Comment by bmwalter68** [comment]
I can confirm that this texture issue happens with just the following installed:

java 1.8.0_51
forge-36.1.17
preview_OptiFine_1.16.5_HD_U_G8_pre14
refinedstorage-1.9.12
refinedstorageaddons-0.7.2
SEUS Renewed Shaders 1.0.1

The rendering issue only appears on cutout blocks that have energy (controller with energy_on=true, grid with connected=true) as well as the disk storage indicator on the disk drive model.  I can confirm it also happens when running with java 11.0.10.  This is clearly a shader issue that crept in between pre2 and pre3 and was partially mitigated in in pre4 or pre5.

pre14: just OptiFine and Refined Storage with Java 1.8.0_51
![2021-05-13_17 18 40](https://user-images.githubusercontent.com/4651048/118190025-893ec200-b410-11eb-8f27-9aa8bc0a8a89.png)

**Comment by bmwalter68** [comment]
I see this is flagged as "Details Needed"  Please advise as to what other details I can add.  I've provided screenshots+F3 and reduced this to just Forge+OptiFine pre14+Refined Storage+Shaders so I'm not sure what else I can do.

**Comment by gri573** [comment]
This issue also occurs with blocks from the mod "Forbidden & Arcanus" that have an emissive overlay (OptiFine G8 full release)

**Comment by bmwalter68** [comment]
I can confirm that the issue with Forbidden and Arcanus follows the same pattern of appearing after pre1 and persisting through G8:

pre1:
![2021-05-26_11 48 38](https://user-images.githubusercontent.com/4651048/119692024-0e66a580-be19-11eb-9215-a1be18f2dda1.png)

G8:
![2021-05-26_11 51 56](https://user-images.githubusercontent.com/4651048/119692052-14f51d00-be19-11eb-8f9f-87764a4b2e86.png)

**Comment by bmwalter68** [comment]
> 
> 
> Have you tried not using optiforge? You can put optifine into the mods folder. I don't think there is an actual need to use optiforge to be honest.

Confirmed this hapens without OptiForge.  Requirements reduced to just:

Forge
OptiFine pre14+
Refined Storage/Forbidden and Arcanus
Shaders

**Comment by gri573** [comment]
I think it may be beneficial to link relevant issues from the affected mods' repositories, both are fixed on the respective mod's side:
[https://github.com/refinedmods/refinedstorage/issues/2919]()
[https://github.com/stal111/Forbidden-Arcanus/issues/165]()

**Comment by KaiAF** [comment]
This looks like it is an issue with refined storage. I would suggest updating the mod. I am going to close this as this doesn't seem to be an optifine issue. If I am wrong, please let me know.

> I see this is flagged as "Details Needed" Please advise as to what other details I can add. I've provided screenshots+F3 and > reduced this to just Forge+OptiFine pre14+Refined Storage+Shaders so I'm not sure what else I can do.

This is because I am just now looking at the issue again.

**Comment by bmwalter68** [comment]
Confirmed as a Refined Storage issue resolved by [this commit](https://github.com/refinedmods/refinedstorage/commit/18d29ec09d33b5162f0c90139f7e8382d11a4fe9)
